Adding 4 Harvard Center for Astronomy datasets.

ado8 opened the pull request:
# Adds 4 supernova datasets to Astropile from Harvard's Center for Astronomy
Collaboration with @tom-hehir @mb010 @erinhay @David-Chemaly @benboyd97 @mattgrayling and others.
Includes CfA 3, CfA 4, CfA Stripped-envelope Core-collapse supernovae, and CfA Type II Supernovae datasets.
The download_data and build_parent_sample python files are shared between all datasets, requiring an argument of cfa3, cfa4, cfa3_4sh, or cfa_snII respectively.

## Links for reference:
[CfA Data Products](https://lweb.cfa.harvard.edu/supernova/) 
- [CfA3 paper](https://ui.adsabs.harvard.edu/abs/2009ApJ...700..331H/abstract)
- [CfA4 paper](https://ui.adsabs.harvard.edu/abs/2012ApJS..200...12H/abstract)
- [CfA Stripped-Envelope Core-collapse paper](https://ui.adsabs.harvard.edu/abs/2014ApJS..213...19B/abstract)
- [CfA Type II paper](https://ui.adsabs.harvard.edu/abs/2017ApJS..233....6H/abstract)

## HuggingFace Dataset Format
Each example contains:

  - `object_id`: Dataset specific identifier
  - `spec_class`: Spectral class
  - `ra`, `dec`: right ascension, declination
  - `lightcurve`: object with:
                - `band`: band of observation, arr(string),
                - `time`: timestamp of observation in MJD, arr(float)
                - `mag`: Magnitude (Standard system) value of observation, arr(float),
                - `mag_err`: Magnitude (Standard system) uncertainty of observation, arr(float)

LTMeyer commented on scripts/.DS_Store:
Should this file be kept in the git history?

LTMeyer commented on download_data.py:
```diff
@@ -0,0 +1,43 @@
+import requests
+import tarfile
+import requests
+import os
```
`os` is imported twice.

LTMeyer commented on download_data.py:
```diff
@@ -0,0 +1,43 @@
+def main(args):
+
+    # Construct the URL to download 
+    url = 'https://raw.githubusercontent.com/djones1040/Foundation_DR1/master/Foundation_DR1/'
```
Is this url gonna be permanently accessible? Is this the official source of data?

LTMeyer commented on scripts/cfa/build_parent_sample.py:
```diff
@@ -0,0 +1,315 @@
+}
+
+
+def main(args):
```
Listing all arguments instead of grouping them into a single `args` would help in reviewing the logic of the function and detecting any unused or unprovided argument.

LTMeyer reviewed (commented):
Great work which mostly looks good to me. I have made few minor comments about the code.

ado8 commented:
I'm not sure where the dot file crept in from, but I removed it.
I'm not great with managing git histories, but I tried getting rid of it via

> git filter-branch --invert-paths --path scripts/.DS_Store --force

but then had trouble pushing.
I deleted my local repo, cloned it again, and just removed the file, so
it's still in the history.

This also applies to the base directory download_data.py, which I think I
pulled in from the YSE or sne_ben branch for reference and forgot to prune.
It's also still in the history but probably shouldn't be.

RE: scripts/cfa/build_parent_sample.py
<https://urldefense.com/v3/__https://github.com/AstroPile/AstroPile_prototype/pull/53*discussion_r1584456526__;Iw!!PvDODwlR4mBZyAb0!Xynqr4Yp96QEEDOmYyo1_fKZ1w9XF6kzG7s_o9b0QA4vH1eqAl2CorlLXCJPXggJEqG16WBy0o_yCe_WlTjWUA$>
:
I unpacked the relevant args attributes into their own variables (e.g.
args.dataset -> dataset) in the main function and passed them explicitly to
the survey_specifc logic functions.
This results in functions that only work for one survey ('cfa3_bpf',
'cfa_snII_bpf') receiving an extra dataset argument so I just added
**kwargs to catch unused function arguments.

On Tue, Apr 30, 2024 at 10:35 AM Lucas Meyer ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Great work which mostly looks good to me. I have made few minor comments
> about the code.
> ------------------------------
>
> On scripts/.DS_Store
> <https://urldefense.com/v3/__https://github.com/AstroPile/AstroPile_prototype/pull/53*discussion_r1584446749__;Iw!!PvDODwlR4mBZyAb0!Xynqr4Yp96QEEDOmYyo1_fKZ1w9XF6kzG7s_o9b0QA4vH1eqAl2CorlLXCJPXggJEqG16WBy0o_yCe9ZzQQ9Uw$>
> :
>
> Should this file be kept in the git history?
> ------------------------------
>
> In download_data.py
> <https://urldefense.com/v3/__https://github.com/AstroPile/AstroPile_prototype/pull/53*discussion_r1584447150__;Iw!!PvDODwlR4mBZyAb0!Xynqr4Yp96QEEDOmYyo1_fKZ1w9XF6kzG7s_o9b0QA4vH1eqAl2CorlLXCJPXggJEqG16WBy0o_yCe9jcjCXvA$>
> :
>
> > @@ -0,0 +1,43 @@
> +import os
> +import argparse
> +import requests
> +import tarfile
> +import requests
> +import os
>
> os is imported twice.
> ------------------------------
>
> In download_data.py
> <https://urldefense.com/v3/__https://github.com/AstroPile/AstroPile_prototype/pull/53*discussion_r1584452198__;Iw!!PvDODwlR4mBZyAb0!Xynqr4Yp96QEEDOmYyo1_fKZ1w9XF6kzG7s_o9b0QA4vH1eqAl2CorlLXCJPXggJEqG16WBy0o_yCe8Cq-As0Q$>
> :
>
> > +            file.write(response.text)
> +    else:
> +        print(f"Failed to download text file. Status code: {response.status_code}")
> +
> +def read_text_file(file_path):
> +    with open(file_path, 'r') as file:
> +        contents = file.read()
> +        return contents
> +
> +
> +
> +
> +def main(args):
> +
> +    # Construct the URL to download
> +    url = 'https://raw.githubusercontent.com/djones1040/Foundation_DR1/master/Foundation_DR1/' <https://urldefense.com/v3/__https://raw.githubusercontent.com/djones1040/Foundation_DR1/master/Foundation_DR1/'__;!!PvDODwlR4mBZyAb0!Xynqr4Yp96QEEDOmYyo1_fKZ1w9XF6kzG7s_o9b0QA4vH1eqAl2CorlLXCJPXggJEqG16WBy0o_yCe_TNrVehg$>
>
> Is this url gonna be permanently accessible? Is this the official source
> of data?
> ------------------------------
>
> In scripts/cfa/build_parent_sample.py
> <https://urldefense.com/v3/__https://github.com/AstroPile/AstroPile_prototype/pull/53*discussion_r1584456526__;Iw!!PvDODwlR4mBZyAb0!Xynqr4Yp96QEEDOmYyo1_fKZ1w9XF6kzG7s_o9b0QA4vH1eqAl2CorlLXCJPXggJEqG16WBy0o_yCe_WlTjWUA$>
> :
>
> > +            data_["FLT"].append(current_filter)
> +        for key in keys_data:
> +            data[key].append(np.array(data_[key]))
> +        f.close()
> +    return num_examples, data, metadata
> +
> +
> +survey_specific_logic = {
> +    "cfa3": cfa3_bpf,
> +    "cfa3_4sh": cfa_generic_bpf,
> +    "cfa4": cfa_generic_bpf,
> +    "cfa_snII": cfa_snII_bpf,
> +}
> +
> +
> +def main(args):
>
> Listing all arguments instead of grouping them into a single args would
> help in reviewing the logic of the function and detecting any unused or
> unprovided argument.
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/AstroPile/AstroPile_prototype/pull/53*pullrequestreview-2030764099__;Iw!!PvDODwlR4mBZyAb0!Xynqr4Yp96QEEDOmYyo1_fKZ1w9XF6kzG7s_o9b0QA4vH1eqAl2CorlLXCJPXggJEqG16WBy0o_yCe-taobtqA$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AHVJWBOKIPGSEI3CJUPMXZDY75QUZAVCNFSM6AAAAABGN67LCSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAMZQG43DIMBZHE__;!!PvDODwlR4mBZyAb0!Xynqr4Yp96QEEDOmYyo1_fKZ1w9XF6kzG7s_o9b0QA4vH1eqAl2CorlLXCJPXggJEqG16WBy0o_yCe9s0WoyXQ$>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

helenqu commented:
no additional comments from me, besides just going through and removing some commented out code (e.g. https://github.com/AstroPile/AstroPile_prototype/pull/53/files#diff-df76b783bac651fcd8c72ba302d550bf91d2e46cdfbba86cd778cbebd0bde681R57)

most importantly, make sure to resolve the merge conflict with the `.gitignore` file, it looks like you just added an extra line which is upsetting git.

ado8 commented:
I think everything is ready to go for this and the CSP branch. Let me know if there's anything else I ought to do.
